### PR TITLE
Assignment for week 9

### PR DESCRIPTION
**Pull request by morarodgers**

### CODE REVIEW

File: models/User.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+USerSchema.pre('save', async function(){
+    const salt = await bcrypt.genSalt(10);
+    this.password = await bcrypt.hash(this.password, salt)
+})
+
+USerSchema.methods.createJWT = function () {
+    return jwt.sign({userID:this._id, name:this.name}, process.env.JWT_SECRET, {expiresIn: process.env.JWT_LIFETIME})
+}
+
+USerSchema.methods.comparePaswsord = async function (candidatePassword) {
```
Comment (joseluiz) [inline models/User.js]: There is a typo here. It should be comparePassword

File: middleware/authentication.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const {UnauthenticatedError} = require('../errors')
+
+
+const auth = (req, res, next) => {
+    // Check header
+    const authHeader = req.headers.authorization
+    if(!authHeader || !authHeader.startsWith('Bearer ')) {
+        throw new UnauthenticatedError('Authentication invalid')
+    }
+    const token = authHeader.split('')[1]
```
Comment (joseluiz) [inline middleware/authentication.js]: There is an error here. You should specify a space to split. Ex:  authHeader.split(' ')[1]

File: controllers/notes.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+const getNote = async(req, res) => {
+    res.send('get one note')
+}
+
+const createNote = async(req, res) => {
+    //res.send('create a job')
+    req.body.createdBy = req.user.userId
+    const note = await Note.create(req.body)
+    res.status(StatusCodes.CREATED).json({job})
```
Comment (joseluiz) [inline controllers/notes.js]: There is an error here. You should return the note var not the job.

File: models/User.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+})
+
+
+USerSchema.pre('save', async function(){
+    const salt = await bcrypt.genSalt(10);
+    this.password = await bcrypt.hash(this.password, salt)
+})
+
+USerSchema.methods.createJWT = function () {
+    return jwt.sign({userID:this._id, name:this.name}, process.env.JWT_SECRET, {expiresIn: process.env.JWT_LIFETIME})
```
Comment (joseluiz) [inline models/User.js]: It should be userId not userID here.

**Review by joseluiz — CHANGES_REQUESTED** [review]
Hi Rodgers. I left some comments of things you need to fix. Please see the comments in the PR for details.

File: controllers/notes.js (hunk 3 above)
Comment (morarodgers) [reply controllers/notes.js]: I have fixed it

File: models/User.js (hunk 4 above)
Comment (morarodgers) [reply models/User.js]: I have fixed it

File: middleware/authentication.js (hunk 2 above)
Comment (morarodgers) [reply middleware/authentication.js]: I have corrected it

**Review by joseluiz — APPROVED** [review]
Hey Rodger. I'm approving the work from week 9. I didn't review the work from week 10. Please keep the work from week 10 in a separate branch. And once you finish create the PR for week 10 for review. ;)

**Pull request merged**